client/my-sites/plugins/plugin-automated-transfer/index.jsx

```diff
@@ -4,110 +4,155 @@
 import React, { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
-import { includes } from 'lodash';
 import wrapWithClickOutside from 'react-click-outside';
 
 /**
  * Internal dependencies
  */
 import { transferStates } from 'state/automated-transfer/constants';
 import { getSelectedSiteId } from 'state/ui/selectors';
-import { getAutomatedTransferStatus } from 'state/automated-transfer/selectors';
+import {
+	getAutomatedTransferStatus,
+	isAutomatedTransferTransferring,
+} from 'state/automated-transfer/selectors';
 import Notice from 'components/notice';
 import NoticeAction from 'components/notice/notice-action';
 
 class PluginAutomatedTransfer extends Component {
 
 	static propTypes = {
+		isTransferring: PropTypes.bool,
 		plugin: PropTypes.object,
-		status: PropTypes.string,
+		transferState: PropTypes.string,
 		translate: PropTypes.func,
-	}
+	};
 
 	static defaultProps = {
 		plugin: {
 			name: 'Plugin',
 		},
-	}
+	};
 
 	state = {
 		clickOutside: false,
+		shouldDisplay: false,
+		transferComplete: false,
+	};
+
+	componentWillMount() {
+		const { COMPLETE, CONFLICTS } = transferStates;
+		const { isTransferring, transferState } = this.props;
+
+		if ( COMPLETE === transferState ) {
+			this.setState( { transferComplete: true } );
+		} else if ( isTransferring || CONFLICTS === transferState ) {
+			this.setState( { shouldDisplay: true } );
+		}
 	}
 
 	componentWillReceiveProps( nextProps ) {
-		if ( this.props.status !== nextProps.status ) {
-			this.setState( { clickOutside: false } );
+		const { COMPLETE, CONFLICTS } = transferStates;
+		const { transferComplete } = this.state;
+		const newState = {};
+
+		if ( this.props.transferState !== nextProps.transferState ) {
+			newState.clickOutside = false;
 		}
+
+		if ( COMPLETE === nextProps.transferState ) {
+			newState.transferComplete = true;
+			if ( ! transferComplete ) {
+				newState.shouldDisplay = true;
+			}
+		} else if ( ! transferComplete ) {
+			if ( nextProps.isTransferring || CONFLICTS === nextProps.transferState ) {
+				newState.shouldDisplay = true;
+			} else {
+				newState.shouldDisplay = false;
```

Comment:
Do you think it'd be too hard to read to do this here instead of the `if/else`? I kind of like how it lessens the amount of lines:

`newState.shouldDisplay = nextProps.isTransferring || CONFLICTS === nextProps.transferState`